Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -624,6 +626,11 @@ namespace Babylon
         const bool normalized = info[7].As<Napi::Boolean>().Value();
 
         vertexArray->RecordVertexBuffer(vertexBuffer, location, byteOffset, byteStride, numElements, type, normalized);
```

Comment:
I think maybe it would be better for RecordVertexBuffer to return a boolean instead of a separate function? It would mean vertexBuffer::CreateHandle will need to return a boolean also, which makes perfect sense.

Also, does index buffer need the same check?

Reply:
Yep, we will need to check index buffer creation as well. It can fail the same way Vertex buffer does. One car argue that vertex buffer creation will fail first, but since we will not prevent the user from creating new Index buffers once they run into an error on creating Vertex buffers, we should handle that for index buffers as well.

Reply:
Updated commit with validation for Index buffer as well. Also, removed the IsValid method from VertexBuffer and made CreateHandle return false if the handler was invalid. However, in order to completely skip binding the vertex buffers and index buffers during draw call it was required to add a IsValid() method to VertexArray, this will be set to false if any associated index of vertex buffer is invalid, this way we completely skip the draw if the VertexArray has invalid data.